homeassistant/components/integration/sensor.py

```diff
@@ -89,6 +90,71 @@
 )
 
 
+class _IntegrationMethod(ABC):
+    @staticmethod
+    def from_name(method_name: str) -> _IntegrationMethod:
+        if method_name == METHOD_TRAPEZOIDAL:
+            return _Trapezoidal()
+        if method_name == METHOD_RIGHT:
+            return _Right()
+        return _Left()
+
+    @abstractmethod
+    def validate_states(
+        self, hass: HomeAssistant, left: State | None, right: State | None
+    ) -> bool:
+        """Check state requirements for integration."""
+
+    @abstractmethod
+    def calculate_area_with_two_states(
+        self, elapsed_time: float, left: State, right: State
+    ) -> Decimal:
+        """Calculate area given two states."""
+
+    def calculate_area_with_one_state(
+        self, elapsed_time: float, constant_state: State
+    ) -> Decimal:
+        return Decimal(constant_state.state) * Decimal(elapsed_time)
+
+
+class _Trapezoidal(_IntegrationMethod):
+    def calculate_area_with_two_states(
+        self, elapsed_time: float, left: State, right: State
+    ) -> Decimal:
+        return Decimal(elapsed_time) * (Decimal(left.state) + Decimal(right.state)) / 2
+
+    def validate_states(
+        self, hass: HomeAssistant, left: State | None, right: State | None
+    ) -> bool:
+        return condition.async_numeric_state(
```

Comment:
I don't like using the `condition.async_numeric_state` here, it's over complicated for what this integration needs.

Maybe add a new helper in `homeassistant/helpers/sensor.py` or `homeassistant/components/sensor/` to help validate a sensor state is numeric? Maybe something can be extracted from `SensorEntity.state`?